### ISSUE TITLE
System Firmware (UEFI) (0.)1.15 not offered by fwupdmgr for ThinkPad X1 Extreme Gen 4

### ISSUE THREAD

**Issue by ginkel** [fwupd/firmware-lenovo#218]
Hi there,

despite being listed [1] on LVFS, the (0.)1.15 firmware update does not seem to be offered by `fwupdmgr update` for my X1E4.

```
$ sudo fwupdmgr update 
Devices with no available firmware updates: 
 • Prometheus IOTA Config
 • Qualcomm Snapdragon X55 5G
 • UEFI Device Firmware
 • UEFI Device Firmware
 • UEFI Device Firmware
 • UEFI Device Firmware
 • UEFI Device Firmware
 • UEFI Device Firmware
 • UEFI dbx
Devices with the latest available firmware version:
 • Embedded Controller
 • MZVL22T0HBLB-00BL7
 • Prometheus
 • System Firmware
 • Unifying Receiver
```

Any ideas?

[1] https://fwupd.org/lvfs/devices/com.lenovo.ThinkPadN40ET.firmware

--

```
$ sudo fwupdmgr get-devices
20Y5CT01WW
│
├─Embedded Controller:
│     Device ID:          2292ae5236790b47884e37cf162dcf23bfcd1c60
│     Summary:            UEFI ESRT device
│     Current version:    0.1.15
│     Minimum Version:    0.1.15
│     Vendor:             Lenovo (DMI:LENOVO)
│     Update State:       Success
│     GUID:               ab5a2de4-75ef-4906-9533-143d2acad54e
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Supported on remote server
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│   
├─MZVL22T0HBLB-00BL7:
│     Device ID:          04e17fcf7d3de91da49a163ffe4907855c3648be
│     Summary:            NVM Express solid state drive
│     Current version:    6L2QGXB7
│     Vendor:             Samsung (NVME:0x144D)
│     Serial Number:      S64SNF0R314779
│     GUIDs:              4d7a2791-106b-5e72-9cfb-8ea3d89f5421 ← NVME\VEN_144D&DEV_A80A&SUBSYS_144DA801&REV_00
│                         310f81b5-6fce-501e-acfb-487d10501e78 ← NVME\VEN_144D&DEV_A80A&SUBSYS_144DA801
│                         60c89aac-f321-515b-b419-3cf02aa9d375 ← NVME\VEN_144D&DEV_A80A&REV_00
│                         bec63ed7-a95f-54fe-b8cc-8e9fee64ba5a ← NVME\VEN_144D&DEV_A80A
│                         75877a5d-2070-57b7-ba80-feb6c937df57 ← SAMSUNG MZVL22T0HBLB-00BL7
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Supported on remote server
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│                         • Signed Payload
│   
├─Prometheus:
│ │   Device ID:          4295296d98b3ba38c72f6baa33d24f03a1d428f6
│ │   Summary:            Fingerprint reader
│ │   Current version:    10.01.3478575
│ │   Vendor:             Synaptics (USB:0x06CB)
│ │   Install Duration:   2 seconds
│ │   Serial Number:      133052029190748
│ │   GUIDs:              d67c4925-e30f-53e6-abc6-e7af0e4c5884 ← USB\VID_06CB&PID_00F9&REV_0000
│ │                       659f7e45-8d45-528d-b3c7-0695eed055f6 ← USB\VID_06CB&PID_00F9
│ │   Device Flags:       • Updatable
│ │                       • Supported on remote server
│ │                       • Cryptographic hash verification is available
│ │                       • Signed Payload
│ │ 
│ └─Prometheus IOTA Config:
│       Device ID:        ea324f9d78d801d381c77ba6947fb8112bebc3a8
│       Summary:          Fingerprint reader config
│       Current version:  0007
│       Minimum Version:  0007
│       Vendor:           Synaptics (USB:0x06CB)
│       GUIDs:            49a675dd-c0d7-5c08-b19a-78fdfb9ea399 ← USB\VID_06CB&PID_00F9-cfg
│                         d2e397c9-fe28-5506-bf0d-b5860baa33ba ← USB\VID_06CB&PID_00F9&CFG1_3692&CFG2_0
│       Device Flags:     • Updatable
│                         • Only version upgrades are allowed
│                         • Signed Payload
│     
├─Qualcomm Snapdragon X55 5G:
│     Device ID:          bf1deb9b7f73ec482561ab8137c35a36e2bb1c75
│     Summary:            Mobile broadband device
│     Current version:    T99W175.F0.1.0.0.9.DT.003.055
│     Vendor:             foxconn (PCI:0x105B)
│     GUIDs:              9624912d-55be-59ca-8f65-91e3d84bc72e ← PCI\VID_105B&PID_E0AB&REV_0000&CARRIER_DT
│                         39fccad9-98bc-5221-83b2-ca390b2c954d ← PCI\VID_105B&PID_E0AB&REV_0000
│                         8e43be37-c679-5557-8309-4cc6ba927dc0 ← PCI\VID_105B&PID_E0AB
│                         c8591d06-63e4-55e7-9dec-3b0783b15e2c ← PCI\VID_105B
│     Device Flags:       • Updatable
│                         • System requires external power source
│   
├─System Firmware:
│ │   Device ID:          a45df35ac0e948ee180fe216a5f703f32dda163f
│ │   Summary:            UEFI ESRT device
│ │   Current version:    0.1.14
│ │   Minimum Version:    0.1.0
│ │   Vendor:             Lenovo (DMI:LENOVO)
│ │   Update State:       Success
│ │   GUIDs:              f1918178-1ab6-4d4e-b82c-26e4d0e3869c
│ │                       230c8b18-8d9b-53ec-838b-6cfc0383493a ← main-system-firmware
│ │   Device Flags:       • Internal device
│ │                       • Updatable
│ │                       • System requires external power source
│ │                       • Supported on remote server
│ │                       • Needs a reboot after installation
│ │                       • Cryptographic hash verification is available
│ │                       • Device is usable for the duration of the update
│ │ 
│ └─UEFI dbx:
│       Device ID:        362301da643102b9f38477387e2193e57abaa590
│       Summary:          UEFI revocation database
│       Current version:  267
│       Minimum Version:  267
│       Vendor:           UEFI:Linux Foundation
│       Install Duration: 1 second
│       GUIDs:            14503b3d-73ce-5d06-8137-77c68972a341 ← UEFI\CRT_A9087D1044AD18F7A94916D284CBC01827CF23CD8F60B79072C9CAA1FEF4D649
│                         5971a208-da00-5fce-b5f5-1234342f9cf7 ← UEFI\CRT_A9087D1044AD18F7A94916D284CBC01827CF23CD8F60B79072C9CAA1FEF4D649&ARCH_X64
│                         c6682ade-b5ec-57c4-b687-676351208742 ← UEFI\CRT_A1117F516A32CEFCBA3F2D1ACE10A87972FD6BBE8FE0D0B996E09E65D802A503
│                         f8ba2887-9411-5c36-9cee-88995bb39731 ← UEFI\CRT_A1117F516A32CEFCBA3F2D1ACE10A87972FD6BBE8FE0D0B996E09E65D802A503&ARCH_X64
│                         5242a674-016e-531a-b134-c8716e311d53 ← UEFI\CRT_A5B488C2ABDC65120351FD1BB9D3A1F295CA6AF098FCB0504879CF5A9E23B8B2
│                         68c16e55-695b-5b3c-9f92-30b6d338a8b4 ← UEFI\CRT_A5B488C2ABDC65120351FD1BB9D3A1F295CA6AF098FCB0504879CF5A9E23B8B2&ARCH_X64
│       Device Flags:     • Internal device
│                         • Updatable
│                         • Needs a reboot after installation
│                         • Only version upgrades are allowed
│                         • Signed Payload
│     
├─UEFI Device Firmware:
│     Device ID:          349bb341230b1a86e5effe7dfe4337e1590227bd
│     Summary:            UEFI ESRT device
│     Current version:    4028499696
│     Minimum Version:    1
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     GUID:               38a375e4-3927-42a2-ab34-e250d748b698
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│   
├─UEFI Device Firmware:
│     Device ID:          f95c9218acd12697af946874bfe4239587209232
│     Summary:            UEFI ESRT device
│     Current version:    16777223
│     Minimum Version:    1
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     GUID:               62009ffc-86f3-4a7c-93a6-1b9b0dd6472d
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│   
├─UEFI Device Firmware:
│     Device ID:          d96de5c124b60ed6241ebcb6bb2c839cb5580786
│     Summary:            UEFI ESRT device
│     Current version:    0
│     Minimum Version:    1
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     GUID:               52c9711e-7a5d-4e8d-ac4a-ba4d920a1008
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│   
├─UEFI Device Firmware:
│     Device ID:          f37fb01122dd62c773f4e84ec89737e059712d59
│     Summary:            UEFI ESRT device
│     Current version:    1
│     Minimum Version:    1
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     GUID:               f18786ab-c122-4f94-a653-a66a88de271e
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│   
├─UEFI Device Firmware:
│     Device ID:          36efb79c255f402f619fa9eb53cd659db51f2a04
│     Summary:            UEFI ESRT device
│     Current version:    1967856
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     GUID:               1d9282eb-f2bf-4c75-8307-028a34978eb9
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│   
├─UEFI Device Firmware:
│     Device ID:          11a7fbdd09ce583e58b899660fe10f147fd155a2
│     Summary:            UEFI ESRT device
│     Current version:    24596
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     GUID:               7c8c8f35-13d3-5271-5e44-5a045511746b
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│   
└─Unifying Receiver:
      Device ID:          3801e1f7043b9abdf1f873bf0bf0b945402e4238
      Summary:            Miniaturised USB wireless receiver
      Current version:    RQR24.11_B0036
      Bootloader Version: BOT03.02_B0009
      Vendor:             HIDRAW:0x046D|USB:0x046D
      Install Duration:   30 seconds
      GUIDs:              cc4cbfa9-bf9d-540b-b92b-172ce31013c1
                          fcf55bf5-767b-51ce-9c17-f6f538c4ee9f ← HIDRAW\VEN_046D&DEV_C52B&REV_00
                          279ed287-3607-549e-bacc-f873bb9838c4 ← HIDRAW\VEN_046D&DEV_C52B
      Device Flags:       • Updatable
                          • Supported on remote server
                          • Signed Payload
$ fwupdmgr --version
runtime   org.freedesktop.fwupd         1.7.7
compile   org.freedesktop.gusb          0.3.10
runtime   com.dell.libsmbios            2.4
runtime   org.kernel                    5.17.4-arch1-1
compile   com.hughsie.libjcat           0.1.10
compile   org.freedesktop.fwupd         1.7.7
runtime   org.freedesktop.gusb          0.3.10
```

**Comment by mrhpearson** [comment]
That's strange - it should work - the GUID matches. Looks like it's had over 1000 downloads too.

Are you plugged in, with battery > 50% and the lid open? 

I just did the update on my P1G4 as I hadn't done it yet...it all went smoothly.

**Comment by ginkel** [comment]
> That's strange - it should work - the GUID matches. Looks like it's had over 1000 downloads too.
> 
> Are you plugged in, with battery > 50% and the lid open?

Plugged in, battery is at 80%, lid open...

Ah, it seems I forgot to `fwupdmgr refresh`. Now I even get offer to update the X55 5G modem:

```
$ sudo fwupdmgr update   
Devices with no available firmware updates: 
 • Prometheus IOTA Config
 • UEFI Device Firmware
 • UEFI Device Firmware
 • UEFI Device Firmware
 • UEFI Device Firmware
 • UEFI Device Firmware
 • UEFI Device Firmware
 • UEFI dbx
 • Unifying Receiver
Devices with the latest available firmware version:
 • Embedded Controller
 • MZVL22T0HBLB-00BL7
 • Prometheus
╔══════════════════════════════════════════════════════════════════════════════╗
║ Upgrade Qualcomm Snapdragon X55 5G from T99W175.F0.1.0.0.9.DT.003.055 to     ║
║ T99W175.F0.1.0.0.9.DT.003.067?                                               ║
╠══════════════════════════════════════════════════════════════════════════════╣
║ Firmware release T99W175.F0.1.0.0.9.DT.003.067 packaged for LVFS             ║
║                                                                              ║
║ This stable release fixes the following issues:                              ║
║                                                                              ║
║ • The release is based on Qualcomm LE1.2 baseline                            ║
║ • Integrated Qualcomm CR#3055483 to fix module lost issue when upgrade       ║
║ firmware image sometimes                                                     ║
║                                                                              ║
║ Qualcomm Snapdragon X55 5G and all connected devices may not be usable       ║
║ while updating.                                                              ║
╚══════════════════════════════════════════════════════════════════════════════╝

Perform operation? [Y|n]: 
```

Is this safe to install, i.e. will the 5G modem continue to work under Linux (in contrast to the firmware versions currently offered on lenovo.com)?

**Comment by mrhpearson** [comment]
Not sure what you mean by 'in contrast to the firmware versions currently offered on lenovo.com' - do they not work? If the issue is FCC unlock related I can't unfortunately help you with that but I hear the modem manager forum has some useful tips.

I'm afraid I also don't have WWAN on any of my systems so limited experience with it. I think that's the first FW update we've offered on LVFS (finally - woohoo!). If it helps - it has 87 downloads with 5 success reports and 1 failure report (which was due to "failed to detach using modem_manager"...not sure how bad that is)

Can we close this bug?

Mark
